jsapp/js/components/library/singleCollectionStore.es6

```diff
@@ -0,0 +1,341 @@
+
+    this.setDefaultColumns();
+
+    hashHistory.listen(this.onRouteChange.bind(this));
+    searchBoxStore.listen(this.searchBoxStoreChanged);
+    actions.library.moveToCollection.completed.listen(this.onMoveToCollectionCompleted);
+    actions.library.subscribeToCollection.completed.listen(this.fetchData.bind(this, true));
+    actions.library.unsubscribeFromCollection.completed.listen(this.fetchData.bind(this, true));
+    actions.library.searchMyLibraryMetadata.completed.listen(this.onSearchMetadataCmpleted);
+    actions.resources.loadAsset.completed.listen(this.onAssetChanged);
+    actions.resources.updateAsset.completed.listen(this.onAssetChanged);
+    actions.resources.cloneAsset.completed.listen(this.onAssetCreated);
+    actions.resources.createResource.completed.listen(this.onAssetCreated);
+    actions.resources.deleteAsset.completed.listen(this.onDeleteAssetCompleted);
+    // These are the changed requests, not all have been updated yet. Trying to
+    // just get assets to show up first
+    actions.library.searchMyCollectionAssets.started.listen(this.onSearchStarted);
+    actions.library.searchMyCollectionAssets.completed.listen(this.onSearchCompleted);
+    actions.library.searchMyCollectionAssets.failed.listen(this.onSearchFailed);
+    // TODO Improve reaction to uploads being finished after task is done:
+    // https://github.com/kobotoolbox/kpi/issues/476
+    actions.resources.createImport.completed.listen(this.fetchDataDebounced);
+
+    // startup store after config is ready
+    actions.permissions.getConfig.completed.listen(this.startupStore);
+  },
+
+  /**
+   * Only makes a call to BE when loaded app on a library route
+   * otherwise wait until route changes to a library (see `onRouteChange`)
+   */
+  startupStore() {
+    if (this.isVirgin && isOnSingleCollectionRoute() && !this.data.isFetchingData) {
+      this.fetchData(true);
+    }
+  },
+
+  setDefaultColumns() {
+    this.data.orderColumnId = this.DEFAULT_ORDER_COLUMN.id;
+    this.data.orderValue = this.DEFAULT_ORDER_COLUMN.defaultValue;
+    this.data.filterColumnId = null;
+    this.data.filterValue = null;
+  },
+
+  // methods for handling search and data fetch
+
+  /**
+   * @return {object} search params shared by all searches
+   */
+  getSearchParams() {
+    const params = {
+      searchPhrase: searchBoxStore.getSearchPhrase(),
+      pageSize: this.PAGE_SIZE,
+      page: this.data.currentPage,
+      uid: getCurrentCollectionUID() // TODO a better way to get UID through listeners?
+    };
+
+    if (this.data.filterColumnId !== null) {
+      const filterColumn = ASSETS_TABLE_COLUMNS.get(this.data.filterColumnId);
+      params.filterProperty = filterColumn.filterBy;
+      params.filterValue = this.data.filterValue;
+    }
+
+    return params;
+  },
+
+  /**** Same code as `myLibraryStore` after this point ****/
+
+  fetchMetadata() {
+    actions.library.searchMyLibraryMetadata(this.getSearchParams());
```

Comment:
This needs to be `searchMyCollectionMetadata` (which you need to set up in `actions/library` and `dataInterface`).

Fetching metadata instead of whole assets is useful for server when we know that a small portion of single asset changed.

Reply:
You will also need to listen to `actions.library.searchMyCollectionMetadata.completed.listen(this.onSearchMetadataCompleted);`